tests/unit/data/test_data_subset.py

```diff
@@ -90,6 +88,29 @@ def test_as_cube_latlon(dataset):
         assert iris.util.array_equal(xcoord.points[[0, -1]], [258000, 344000])
         assert iris.util.array_equal(ycoord.points[[0, -1]], [56000, 146000])
 
+
+class TestTrackSubset:
+    @staticmethod
+    @pytest.fixture
+    def dataset(sampledir):
+        path = os.path.join(sampledir, "aircraft", "M285_sample.nc")
+        sample_cube = iris.load_cube(path, 'NO2_concentration_ug_m3')
+        return DataSubset(sample_cube)
+
+    @staticmethod
+    def test_as_cube(dataset):
+        cube = dataset.extract_track('13:00', '14:00')
+
+        # Check we have the correct time bounds (in hours since epoch)
+        assert cube.coord('index').points[0] == int(449197)
+        assert cube.coord('index').points[-1] == int(449198)
+
+    @staticmethod
+    def test_time_bound_error(dataset):
+        with pytest.raises(ValueError, match='Empty dataframe, likely due to time bounds being out of range'):
+            dataset.extract_track('20:00', '21:00')
+
+
```

Comment:
Good tests, but I would like to see a description at the top of every test please.

Reply:
Should we agree between us on a consistent style to use overall? Some have simple 1 line description, a few others use 'GIVEN...WHEN...THEN...' structure. I suppose the latter is better, but can be tricky for some of them.

Reply:
I agree.  I think for all new tests we should use the 'GIVEN... WHEN... THEN...' structure, or as close as we can get to it, and if we have time we can convert old tests to the same or similar format.  Still a bit of a nice-to-have though.